src/client/client.cpp

```diff
@@ -1789,6 +1792,60 @@ float Client::getCurRate()
 			m_con->getLocalStat(con::CUR_DL_RATE));
 }
 
+void Client::storeInventoryImages(u16 res)
+{
+	irr::video::IVideoDriver *driver = RenderingEngine::get_video_driver();
+	infostream<<"- Saving inventory images"<<std::endl;
+	if (driver->queryFeature(video::EVDF_RENDER_TO_TARGET))
+	{
+		std::string screenshot_dir;
+		if (fs::IsPathAbsolute(g_settings->get("screenshot_path")))
+			screenshot_dir = g_settings->get("screenshot_path");
+		else
+			screenshot_dir = porting::path_user + DIR_DELIM + g_settings->get("screenshot_path");
+		std::string inv_img_dir = screenshot_dir + DIR_DELIM + "inventory_images";
+		fs::CreateDir(inv_img_dir);
+		std::set<std::string> result;
+		m_itemdef->getAll(result);
+		for (const std::string item_name : result)
```

Comment:
const std::string &item_name & multple code style issues

Reply:
> multple code style issues

Please elaborate, I haven't done that much C++, so I'd appreciate some pointers.

Reply:
you are copying the string instead of using a ref which is inefficient in terms of performance